### PR TITLE
feat(aapd-323): add Address class

### PR DESCRIPTION
**Pull request by edpackard-work**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/common/src/lib/address.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	/**
+	 * @type {string} - the name of the town, city or other settlement
+	 */
+	town;
+
+	/**
+	 * @type {string} - the name of the county
+	 */
+	county;
```
Comment (chrishawldev) [inline packages/common/src/lib/address.js:24]: Did we have a separate county field? I thought we just had 4 fields. If so then all good.

File: packages/common/src/lib/address.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	/**
+	 * @type {string} - the second line of the address
+	 */
+	addressLine2;
+
+	/**
+	 * @type {string} - the name of the town, city or other settlement
+	 */
+	town;
```
Comment (chrishawldev) [inline packages/common/src/lib/address.js:19]: Maybe call this CityTown as its value could be either

File: packages/common/src/lib/address.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	 * @param {string} params.addressLine2
+	 * @param {string} params.town
+	 * @param {string} params.county
+	 * @param {string} params.postcode
+	 */
+	constructor({ addressLine1, addressLine2, town, county, postcode }) {
+		this.addressLine1 = addressLine1;
+		this.addressLine2 = addressLine2;
+		this.town = town;
+		this.county = county;
```
Comment (chrishawldev) [inline packages/common/src/lib/address.js:44]: I think the other thing we talked about was that addressLine2 and postcode are optional so we could make them the last 2 constructor parameters and then also maybe thrown an error if addressLine1 and City / Town aren't provided (and maybe county if we need that?)

File: packages/common/src/lib/address.js (hunk 1 above)
Comment (edpackard-work) [reply packages/common/src/lib/address.js:24]: oops, I probably got this from looking at addresses on the appellant journey - if we don't need county, I will eliminate it

**Review by chrisHowarthkc — APPROVED** [review]

**Review by chrishawldev — APPROVED** [review]
LGTM

**Pull request merged**